File: CS/Form1.cs
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Data;

namespace Example {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            DataTable table = CreateTable();
            CustomButtonEdit.RepositoryItemCustomButtonEdit edit = new CustomButtonEdit.RepositoryItemCustomButtonEdit();
            edit.GetButtonState += buttonConditionHandler;
            this.gridControl1.RepositoryItems.Add(edit);
            this.gridControl1.DataSource = table;
            this.gridView1.Columns[2].ColumnEdit = edit;
            this.gridView1.OptionsView.ShowButtonMode = DevExpress.XtraGrid.Views.Base.ShowButtonModeEnum.ShowAlways;
        }
        private DataTable CreateTable() {
            DataTable table = new DataTable();
            table.Columns.Add("First column");
            table.Columns.Add("Condition column", typeof(bool));
            table.Columns.Add("Result column");
            table.Rows.Add("Row One", true, "ButtonEdit1");
            table.Rows.Add("Row Two", false, "ButtonEdit2");
            table.Rows.Add("Row Three", false, "ButtonEdit3");
            table.Rows.Add("Row Four", true, "ButtonEdit4");
            table.Rows.Add("Row Five", true, "ButtonEdit5");
            return table;
        }
        private void buttonConditionHandler(object sender, CustomButtonEdit.ButtonStateEventArgs e) {
            object value = e.IsEnable;
            if (e.Tag != null) {
                GridCellInfo gci = (e.Tag as GridCellInfo);
                value = (gci.Column.View as GridView).GetRowCellValue(gci.RowHandle, (gci.Column.View as GridView).Columns[1]);
            }
            else if (gridView1.ActiveEditor == sender) {
                value = gridView1.GetRowCellValue(gridView1.FocusedRowHandle, gridView1.Columns[1]);
            }
            e.IsEnable = (value != null && value.GetType() == typeof(bool)) ? (bool)value : false;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Data

Namespace Example
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim table As DataTable = CreateTable()
			Dim edit As New CustomButtonEdit.RepositoryItemCustomButtonEdit()
			AddHandler edit.GetButtonState, AddressOf buttonConditionHandler
			Me.gridControl1.RepositoryItems.Add(edit)
			Me.gridControl1.DataSource = table
			Me.gridView1.Columns(2).ColumnEdit = edit
			Me.gridView1.OptionsView.ShowButtonMode = DevExpress.XtraGrid.Views.Base.ShowButtonModeEnum.ShowAlways
		End Sub
		Private Function CreateTable() As DataTable
			Dim table As New DataTable()
			table.Columns.Add("First column")
			table.Columns.Add("Condition column", GetType(Boolean))
			table.Columns.Add("Result column")
			table.Rows.Add("Row One", True, "ButtonEdit1")
			table.Rows.Add("Row Two", False, "ButtonEdit2")
			table.Rows.Add("Row Three", False, "ButtonEdit3")
			table.Rows.Add("Row Four", True, "ButtonEdit4")
			table.Rows.Add("Row Five", True, "ButtonEdit5")
			Return table
		End Function
		Private Sub buttonConditionHandler(ByVal sender As Object, ByVal e As CustomButtonEdit.ButtonStateEventArgs)
			Dim value As Object = e.IsEnable
			If e.Tag IsNot Nothing Then
				Dim gci As GridCellInfo = (TryCast(e.Tag, GridCellInfo))
				value = (TryCast(gci.Column.View, GridView)).GetRowCellValue(gci.RowHandle, (TryCast(gci.Column.View, GridView)).Columns(1))
			ElseIf gridView1.ActiveEditor Is sender Then
				value = gridView1.GetRowCellValue(gridView1.FocusedRowHandle, gridView1.Columns(1))
			End If
			e.IsEnable = If((value IsNot Nothing AndAlso value.GetType() Is GetType(Boolean)), CBool(value), False)
		End Sub
	End Class
End Namespace
